### PR TITLE
Update TF to 2.12 (backport from IB/CMSSW_13_2_X/g13 branch)

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: protobuf.spec [hunk 1]
```diff
@@ -8,25 +8,30 @@
...
 Requires: zlib
 BuildRequires: cmake ninja
 
 %prep
 %setup -n %{n}-%{realversion}
-%patch0 -p1
-sed -i -e 's|CMAKE_CXX_STANDARD  *11|CMAKE_CXX_STANDARD 17|' cmake/CMakeLists.txt
+#%patch0 -p1
+#%patch1 -p1
+#sed -i -e 's|CMAKE_CXX_STANDARD  *11|CMAKE_CXX_STANDARD 17|' cmake/CMakeLists.txt
```
Comment (smuzaffar) [inline protobuf.spec]: @iarspider , why we are not changing the cxx standard? I think default is still `11` ... right?

File: protobuf.spec [hunk 2]
```diff
@@ -8,25 +8,30 @@
 # protoc --cpp_out=. DQMServices/Core/src/ROOTFilePB.proto
 #######################################################
 
-Source: https://github.com/google/protobuf/archive/v%{realversion}.tar.gz
-Patch0: protobuf-3.15-gcc10
+%define keep_archives true
+
+Source: https://github.com/protocolbuffers/protobuf/archive/v3.21.9.zip
+#Patch0: protobuf-3.15-gcc10
+#Patch1: protobuf-std_iterator-8741
```
Comment (smuzaffar) [inline protobuf.spec]: so we do not need these patches?

File: grpc.spec [hunk 3]
```diff
@@ -1,6 +1,7 @@
-### RPM external grpc 1.35.0
+### RPM external grpc 1.33.1-patches
```
Comment (smuzaffar) [inline grpc.spec]: why rollback?

**Comment by iarspider** [comment]
Thanks for the comments, I will double check these issues.

**Comment by iarspider** [comment]
please test

File: protobuf.spec (hunk 1 above)
Comment (smuzaffar) [reply protobuf.spec]: ping

**Comment by smuzaffar** [comment]
please test

@iarspider , are you still working on it?

**Comment by iarspider** [comment]
Yes, running a local test now.

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
test parameters:
- full_cmssw = true

**Comment by iarspider** [comment]
please test

File: protobuf.spec [hunk 4]
```diff
@@ -40,3 +41,4 @@ ninja -v %{makeprocesses}
 cd ../build
 ninja -v %{makeprocesses} install
 rm -rf %{i}/lib/pkgconfig
+ # rebuild #
```
Comment (iarspider) [inline protobuf.spec]: TODO: remove this line before merging!

**Comment by iarspider** [comment]
please test with cms-sw/cmssw#42228

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test with https://github.com/cms-sw/cmssw/pull/42228

**Comment by iarspider** [comment]
please test with https://github.com/cms-sw/cmssw/pull/42228

**Comment by iarspider** [comment]
please test with https://github.com/cms-sw/cmssw/pull/42228

**Comment by iarspider** [comment]
please abort

**Comment by iarspider** [comment]
test parameters:
- pull_request = https://github.com/cms-sw/cmssw/pull/42228
- full_cmssw = true
- enable = profiling

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
please test for el8_aarch64_gcc11

**Comment by iarspider** [comment]
please test for el8_ppc64le_gcc11

**Comment by iarspider** [comment]
please test for el8_aarch64_gcc11

**Comment by iarspider** [comment]
enable profiling

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
BoringSSL refuses to add support PowerPC (both LE and BE). We can try using OpenSSL instead of BoringSSL - testing this locally.

**Comment by iarspider** [comment]
please test for el8_aarch64_gcc11

**Comment by iarspider** [comment]
please test for el8_ppc64le_gcc11

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
PowerPC failures:

* testConfigDP: timed out, shouldn't be related to this PR
* RelVals: segfault in cling::IncrementalExecutor::runStaticInitializersOnce, https://github.com/cms-sw/cmssw/issues/41658

**Comment by iarspider** [comment]
please test

**Comment by smuzaffar** [comment]
please test

lets test for 13.3.X

**Comment by smuzaffar** [comment]
please test for el8_aarch64_gcc11

**Comment by smuzaffar** [comment]
please test for el8_ppc64le_gcc11

**Comment by iarspider** [comment]
please test for el8_ppc64le_gcc11

got stuck, let's retry

**Comment by smuzaffar** [comment]
please test

**Comment by iarspider** [comment]
please test

download failure

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@cms-sw/reconstruction-l2 , can you please check the DQM/reco comparison results here? We are trying to update Tensorflow to version 2.12 and looks like there are few differences in the tracking part. If these changes looks reasonable then we would move forward with TF 2.12 integration ( which also has some fixes for c++20 standard)

**Review by smuzaffar — APPROVED** [review]

**Comment by smuzaffar** [comment]
+externals

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
merge

File: fasthadd.spec [hunk 5]
```diff
@@ -1,10 +1,11 @@
-### RPM external fasthadd 2.3
+### RPM external fasthadd 2.4
 
 #Change the commit hash every time a new version is needed.
-%define commit 905af02d3369428df677c232537da6fca8982ff3
-Source0: https://raw.githubusercontent.com/cms-sw/cmssw/%commit/DQMServices/Components/bin/fastHadd.cc
-Source1: https://raw.githubusercontent.com/cms-sw/cmssw/%commit/DQMServices/Core/src/ROOTFilePB.proto
-Requires: protobuf root
+%define commit 003102d8516b652d80e76d8ca1c4b5dc47c9015b
+%define user iarspider
```
Comment (smuzaffar) [inline fasthadd.spec:5]: @iarspider , I think now we can get the updated files from cmssw repo ... right? can you please open a Pr with change

**Comment by fwyzard** [comment]
This may be causing a failure in the GPU tests, reported here: https://github.com/cms-sw/cmssw/issues/42460 .

**Pull request merged**